src/main/java/org/jboss/modules/ConcurrentClassLoader.java

```diff
@@ -455,7 +455,12 @@ protected Package definePackage(final String name, final String specTitle, final
             if (existing != null) {
                 return existing;
             }
-            Package pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            Package pkg;
+            try {
+                pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            } catch (final IllegalArgumentException iae) {
+                pkg = super.getPackage(name);
+            }
```

Comment:
Is there any chance an IllegalArgumentException can be thrown without the package being defined in the superclass?